cmd/common/utils/parser-tableformatter.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/columns"
+	"github.com/kinvolk/inspektor-gadget/pkg/columns/formatter/textcolumns"
+)
+
+const (
+	KubernetesTag       string = "kubernetes"
+	ContainerRuntimeTag string = "runtime"
+)
+
+var SupportedTags = map[string]struct{}{
+	KubernetesTag:       {},
+	ContainerRuntimeTag: {},
+}
+
+func WithMetadata[T any](tag string) func(*GadgetParser[T]) error {
+	return func(parser *GadgetParser[T]) error {
+		if _, ok := SupportedTags[tag]; !ok {
+			return fmt.Errorf("unknown value for metadata: %q", tag)
+		}
+
+		// The resulting set of columns will be the gadget-specific columns plus
+		// the ones with the requested tag.
+		parser.columnsFilter = columns.Or(columns.WithNoTags(), columns.WithTag(tag))
```

Comment:
I think that's the best way without having to know any field specifics.

Reply:
Applied!

Reply:
Isn't it that bad to rely the way things are defined in the event struct? IMO it's just fine to suppose runtime/kubernetes tags are defined first.

Reply:
Mmmm ensuring the order was achieved by adding these 3 lines:

https://github.com/kinvolk/inspektor-gadget/blob/841b0d8f511bc879385e1ae13102f409ca2d3fd8/cmd/common/utils/parser-tableformatter.go#L84-L88

However, I just realized that if we configure a specific set of columns through `textcolumns.WithDefaultColumns()`, it will also include the columns with `hide` (e.g. the mntns columns in exec and signal gadgets). Therefore, I will remove these lines for now. If we find a problem with the order, we know how what we need to do.